rasa/shared/core/slots.py

```diff
@@ -270,22 +335,31 @@ def as_feature(self) -> List[float]:
             return r
         return r
 
-    def feature_dimensionality(self) -> int:
+    def _feature_dimensionality(self) -> int:
         return len(self.values)
 
 
-class DataSlot(Slot):
+class AnySlot(Slot):
+    """Slot which can be used to store any value. Users need to create a subclass of
+       `Slot` in case the information is supposed to get featurized."""
+
+    type_name = "any"
+
     def __init__(
         self,
         name: Text,
         initial_value: Any = None,
-        value_reset_delay: Optional[int] = 1,
+        value_reset_delay: Optional[int] = None,
         auto_fill: bool = True,
-    ):
-        super().__init__(name, initial_value, value_reset_delay, auto_fill)
+        influence_conversation: bool = False,
+    ) -> None:
+        if influence_conversation:
+            raise ValueError(
+                f"An {AnySlot.__name__} cannot be featurized. "
+                f"Please use a different slot type instead. See the "
```

Comment:
should we hint on being able to subclass this slot if they want to? otherwise that hint is only in the class docstring of `AnySlot`, which is very hidden for someone seeing this warning

Reply:
I think they shouldn't subclass this one but rather `Slot` directly. There is no benefit in subclassing this one. Will add it to the error message 👍